### PR TITLE
Bugfix: Make storage users mount ids unique by default

### LINKED ISSUE #5064: storage uuids are not randomized with init
Currently every deployment uses tha same uuid for the users storage ... that is a non starter for a federated deployment.

We have to initialize the Storageid for the users storage (which in fact currently also is our project storage ... so it is already misused, but that is a separate issue) so that it is possible to determine the storageprovider and distinguish an which instance a space is located ...

https://github.com/owncloud/ocis/blob/da1c18afd8934db1fe7cde89e1e0e9197f33a6c0/services/gateway/pkg/revaconfig/config.go#L130 needs to be made configurabe and set to a random uuid with init

### All WebDAV urls will change

The providerID is the first part of the ID string in the url. It will change from the existing hardcoded uuid (`1284d238-aa92-42ce-bdc4-0b0000009157`) to a new random uuid.

```diff
- "webDavUrl": "https://localhost:9200/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157$4c510ada-c86b-4815-8820-42cdf82c3d51"
+ "webDavUrl": "https://localhost:9200/dav/spaces/bc631809-54a9-4b1e-9d6e-d9bdf5088eab$4c510ada-c86b-4815-8820-42cdf82c3d51"
```
#### Consequences for Clients

- Web: Can handle it
- iOS: Can handle it
- Android: not known
- Desktop: All sync connections need to be recreated

### Alternative Solution

- Fallback to the existing mount ID
- This is not a good approach IMO, we really risk to have a lot of storage providers out there with the same ID
- Later migrations will be even more painful

### PR DESCRIPTION
**Pull request by micbar**
## Description

The mount ID of the storage users provider needs to be unique by default. We made this value configurable and added it to ocis init to be sure that we have a random uuid v4. This is important for federated instances.

## Related Issue

- Fixes #5064 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/storage-users/pkg/config/defaults/defaultconfig.go [hunk 1]
```diff
@@ -39,7 +39,6 @@ func DefaultConfig() *config.Config {
 		},
 		Reva:             shared.DefaultRevaConfig(),
 		DataServerURL:    "http://localhost:9158/data",
-		MountID:          "1284d238-aa92-42ce-bdc4-0b0000009157",
```
[the remark this answers is not in the document]
Comment (micbar) [reply services/storage-users/pkg/config/defaults/defaultconfig.go:42]: good question. Locally it was no problem to change the configuration.

Needs qualification with the clients teams.

[the remark this answers is not in the document]
Comment (micbar) [reply services/storage-users/pkg/config/defaults/defaultconfig.go:42]: I added more information to https://github.com/owncloud/ocis/issues/5064

**Review by butonic — APPROVED** [review]

**Comment by butonic** [comment]
starting an existing instance fails with an error message that explains what to do. we need to break this now so we dont have to break instances later.

**Comment by ScharfViktor** [comment]
I tested it: 
- after init command we have `storage_users_mount_id: 7fa81a2a-b94b-4b71-ac7d-bb30b3fb61f3` and `mount_id: 7fa81a2a-b94b-4b71-ac7d-bb30b3fb61f3` in the `ocis.yaml`. 

My questions: 
- should we put info to the doc?
- `GATEWAY_STORAGE_USERS_MOUNT_ID` and `STORAGE_USERS_MOUNT_ID` are configurable? Can I generate UUID and start init command with envs `GATEWAY_STORAGE_USERS_MOUNT_ID=uuid1` and `STORAGE_USERS_MOUNT_ID=uuid2` ? Does not work for me: `ocis.yaml` contains random uuid, storage_users_mount_id and mount_id are same value

**Pull request merged**